eegprep/utils/utils.py

```diff
@@ -1,6 +1,118 @@
 
 
-def validate_input_dir(input_dir):
+
+def validate_sourcedata(path, source_type, pattern='sub-\\d+'):
+    """
+    This function validates the "sourcedata/" directory provided by user to
+    see if it's contents are consistent with BIDS.
```

Comment:
I'm not sure if the term `BIDS` should be used, as this proposed validator operates on the `sourcedata/` in order to check if everything is ok for a later conversion to `BIDS`, no?

Reply:
Ok, yes, I see your point. However, BIDS does have specific rules (or suggestions) for `sourcedata/` doesn’t it?

Reply:
It does, as you can check/evaluate here: https://bids-specification.readthedocs.io/en/stable/02-common-principles.html#source-vs-raw-vs-derived-data. However, we would need to discuss how this applies to our case.

Reply:
Right, right, let’s kick it out for now. We can always com back and adapt the warning messages if necessary